### PR TITLE
Set correct minimal versions in default conf (development)

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Set `MBEDTLS_SSL_MIN_MAJOR_VERSION` and `MBEDTLS_SSL_MIN_MINOR_VERSION`
instead of `MBEDTLS_SSL_MAJOR_VERSION_3` and `MBEDTLS_SSL_MINOR_VERSION_1`

Rebased #939 to current development to get CI feedback.

Fixes https://github.com/ARMmbed/mbedtls/issues/664

Internal ref: IOTSSL-1387

Backport to 2.1: #953 (not applicable to 1.3)

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
I rebased #939 (single commit 19cd9602ee039c7c09f55c1868a64673bef14b01) to get 5e9f14d4d988222aacb49b8eb15eedd266e9f147 so as to get CI results with modern test scripts. The diff is identical apart from context in the ChangeLog. I also fixed “Bugs” → “Bugfix” in the changelog.

**Comment by gilles-peskine-arm** [comment]
`compat.sh` failed in the `full` config.
```
21:21:18 m->m dtls1_2,yes TLS-ECDH-ECDSA-WITH-AES-128-GCM-SHA256 ................ FAIL
```

**Comment by gilles-peskine-arm** [comment]
retest

**Comment by gilles-peskine-arm** [comment]
In a [second test run](https://jenkins-internal.mbed.com/job/mbedtls-pr/661/execution/node/127/log/), that `compat.sh` run succeeded, the only failure was a FreeBSD timing self-test failure which is a known issue.

File: ChangeLog [hunk 1]
```diff
@@ -41,17 +41,6 @@ New deprecations
...
-   * Extend RSA interface by multiple functions allowing structure-
-     independent setup and export of RSA contexts. Most notably,
-     mbedtls_rsa_import and mbedtls_rsa_complete are introduced for setting
-     up RSA contexts from partial key material and having them completed to the
-     needs of the implementation automatically. This allows to setup private RSA
-     contexts from keys consisting of N,D,E only, even if P,Q are needed for the
-     purpose or CRT and/or blinding.
-   * The configuration option MBEDTLS_RSA_ALT can be used to define alternative
-     implementations of the RSA interface declared in rsa.h.
-
```
Comment (Patater) [inline ChangeLog]: I'm not sure the rebasing went so well. I wouldn't expect to see the `API Changes` section moved if based on a recent enough branch (one that has 2.7.0 in the ChangeLog, at least).

Comment (gilles-peskine-arm) [reply ChangeLog]: It was rebased to some point during 2.7 development. We'll merge the changelog when we merge the PR, as usual, that's not a problem.

**Comment by gilles-peskine-arm** [comment]
Note to gatekeepers: this needs to be merged into 2.7 as well as development. No need for a separate PR since this was forked before 2.7.

**Comment by gilles-peskine-arm** [comment]
Assigning to @yanesca for design approval. If the design is approved, this can be merged (into development and 2.7, and merge the 2.1 backport #953 as well).

**Comment by yanesca** [comment]
Looks good to me!

**Pull request merged**